src/xrpld/rpc/handlers/LedgerEntry.cpp

```diff
@@ -66,768 +67,819 @@ parseAuthorizeCredentials(Json::Value const& jv)
+        return std::nullopt;
+    }
+    try
+    {
+        auto const mptIssuanceIdStr = mptJson[jss::mpt_issuance_id].asString();
+
+        uint192 mptIssuanceID;
+        if (!mptIssuanceID.parseHex(mptIssuanceIdStr))
+            Throw<std::runtime_error>("Cannot parse mpt_issuance_id");
+
+        auto const account =
+            parseBase58<AccountID>(mptJson[jss::account].asString());
+
+        if (!account || account->isZero())
         {
-            expectedType = ltMPTOKEN;
-            auto const& mptJson = context.params[jss::mptoken];
-            if (!mptJson.isObject())
-            {
-                if (!uNodeIndex.parseHex(mptJson.asString()))
-                {
-                    uNodeIndex = beast::zero;
-                    jvResult[jss::error] = "malformedRequest";
-                }
-            }
-            else if (
-                !mptJson.isMember(jss::mpt_issuance_id) ||
-                !mptJson.isMember(jss::account))
-            {
-                jvResult[jss::error] = "malformedRequest";
-            }
-            else
-            {
-                try
-                {
-                    auto const mptIssuanceIdStr =
-                        mptJson[jss::mpt_issuance_id].asString();
-
-                    uint192 mptIssuanceID;
-                    if (!mptIssuanceID.parseHex(mptIssuanceIdStr))
-                        Throw<std::runtime_error>(
-                            "Cannot parse mpt_issuance_id");
-
-                    auto const account = parseBase58<AccountID>(
-                        mptJson[jss::account].asString());
-
-                    if (!account || account->isZero())
-                        jvResult[jss::error] = "malformedAddress";
-                    else
-                        uNodeIndex =
-                            keylet::mptoken(mptIssuanceID, *account).key;
-                }
-                catch (std::runtime_error const&)
-                {
-                    jvResult[jss::error] = "malformedRequest";
-                }
-            }
+            jvResult[jss::error] = "malformedAddress";
+            return std::nullopt;
         }
-        else
+
+        return keylet::mptoken(mptIssuanceID, *account).key;
+    }
+    catch (std::runtime_error const&)
+    {
+        jvResult[jss::error] = "malformedRequest";
+        return std::nullopt;
+    }
+}
+
+using FunctionType =
+    std::function<std::optional<uint256>(Json::Value const&, Json::Value&)>;
+
+struct LedgerEntry
+{
+    Json::StaticString fieldName;
+    FunctionType parseFunction;
+    LedgerEntryType expectedType;
+};
+
+// {
+//   ledger_hash : <ledger>
+//   ledger_index : <ledger_index>
+//   ...
+// }
+Json::Value
+doLedgerEntry(RPC::JsonContext& context)
+{
+    std::shared_ptr<ReadView const> lpLedger;
+    auto jvResult = RPC::lookupLedger(lpLedger, context);
+
+    if (!lpLedger)
+        return jvResult;
+
+    static LedgerEntry ledgerEntryParsers[] = {
```

Comment:
nit: Consider using `std::array` over C-arrays

Reply:
Fixed in https://github.com/XRPLF/rippled/pull/5199/commits/3d5ef0fb972cf519f9c890126e2364e5316a014e - not sure I like it, though